packages/chat-core/src/index.ts

```diff
@@ -221,6 +221,20 @@ export const shouldReinitialize = (
   );
 };
 
+const getLastExpirationTimestamp = (
+  responses: Response[],
+): number | undefined => {
+  const lastExpirationResponse = findLast(
+    (response) =>
+      response.type === "bot" &&
+      typeof response.payload.expirationTimestamp === "number",
+    responses,
+  );
+  return lastExpirationResponse?.type === "bot"
+    ? lastExpirationResponse.payload.expirationTimestamp
+    : undefined;
```

Comment:
minor nit (and less precise but)
```suggestion
return lastExpirationResponse?.payload?.expirationTimestamp;
```

Reply:
I would rather do more precise and a bit clearer on the data structure.